scripts/core/editor3/reducers/spellchecker.jsx

```diff
@@ -28,28 +24,47 @@ const spellchecker = (state = {}, action) => {
 const replaceWord = (state, text) => {
     const {editorState, spellcheckerMenu} = state;
     const {word, editorSelection} = spellcheckerMenu;
-
-    var textSelection;
-    var newContentState;
-    var newEditorState;
-
-    textSelection = editorSelection.merge({
+    const sel = editorSelection.merge({
         anchorOffset: word.offset,
         focusOffset: word.offset + word.text.length
     });
 
-    newContentState = Modifier.replaceText(editorState.getCurrentContent(), textSelection, text);
-    newEditorState = EditorState.push(editorState, newContentState, 'spellchecker');
+    var newState = Modifier.replaceText(editorState.getCurrentContent(), sel, text);
 
-    return Object.assign({}, state, {
-        editorState: newEditorState,
-        spellcheckerMenu: null
-    });
+    newState = EditorState.push(editorState, newState, 'spellchecker');
+
+    return {
+        ...state,
+        editorState: newState,
+        spellcheckerMenu: {
+            ...spellcheckerMenu,
+            visible: false
+        }
+    };
 };
 
 /**
  * @ngdoc method
- * @name Spellchecker Reducers#showContextMenu
+ * @name closeContextMenu
+ * @param {Object} state
+ * @return {Object} returns new state
+ * @description Save on store all relevant data for spellchecker context menu
+ */
+const closeContextMenu = (state) => {
```

Comment:
I think I'm lost here. This component (the context menu) is part of the editor as a whole and the editor should be aware of it.

Your suggestion raises a couple of difficulties which could potentially complicate things even more:
- If each span manages it's own menu and shows it on right click, how would other span's know to close their menu because another one has opened somewhere else?
- Whenever right-clicking a span there has to be a service call to the spellchecker to get the suggestions, which would have to be an action if we are using Redux. When 10-100+ spans are listening to this action, how do they know which one the incoming suggestions are for? Do we use IDs? This seems super complicated. We could potentially leave this out of Redux and let each decorator do the call in the React component itself, but then we aren't using Redux anymore and might confuse things even more.
- What about other actions that might affect the visibility of the context menu? Let's say the "context menu" of the link editing functionality. It seems to me that the editor should be able to manage it's menus.

I'm not sure if we should isolate this functionality...